src/ol/reproj/reproj.js

```diff
@@ -0,0 +1,339 @@
+    // This is not possible if crossing both "dateline" and "prime meridian".
+    var triangulationSrcExtent = triangulation.calculateSourceExtent();
+    var triangulationSrcWidth = goog.math.modulo(
+        ol.extent.getWidth(triangulationSrcExtent), wrapXShiftDistance);
+
+    if (triangulationSrcWidth < wrapXShiftDistance / 2) {
+      wrapXType = ol.reproj.WrapXRendering_.STITCH_SHIFT;
+    } else {
+      wrapXType = ol.reproj.WrapXRendering_.STITCH_EXTENDED;
+    }
+  }
+
+  var srcDataExtent = ol.extent.createEmpty();
+  goog.array.forEach(sources, function(src, i, arr) {
+    if (wrapXType == ol.reproj.WrapXRendering_.STITCH_SHIFT) {
+      var srcW = src.extent[2] - src.extent[0];
+      var srcX = goog.math.modulo(src.extent[0], wrapXShiftDistance);
+      ol.extent.extend(srcDataExtent, [srcX, src.extent[1],
+                                       srcX + srcW, src.extent[3]]);
+    } else {
+      ol.extent.extend(srcDataExtent, src.extent);
+    }
+  });
+  if (!goog.isNull(sourceExtent)) {
+    if (wrapXType == ol.reproj.WrapXRendering_.NONE) {
+      srcDataExtent[0] = goog.math.clamp(
+          srcDataExtent[0], sourceExtent[0], sourceExtent[2]);
+      srcDataExtent[2] = goog.math.clamp(
+          srcDataExtent[2], sourceExtent[0], sourceExtent[2]);
+    }
+    srcDataExtent[1] = goog.math.clamp(
+        srcDataExtent[1], sourceExtent[1], sourceExtent[3]);
+    srcDataExtent[3] = goog.math.clamp(
+        srcDataExtent[3], sourceExtent[1], sourceExtent[3]);
+  }
+
+  var srcDataWidth = ol.extent.getWidth(srcDataExtent);
+  var srcDataHeight = ol.extent.getHeight(srcDataExtent);
+  var canvasWidthInUnits;
+  if (wrapXType == ol.reproj.WrapXRendering_.STITCH_EXTENDED) {
+    canvasWidthInUnits = 2 * wrapXShiftDistance;
+  } else {
+    canvasWidthInUnits = srcDataWidth;
+  }
+
+  var stitchContext = ol.dom.createCanvasContext2D(
+      Math.round(pixelRatio * canvasWidthInUnits / sourceResolution),
+      Math.round(pixelRatio * srcDataHeight / sourceResolution));
+
+  stitchContext.scale(pixelRatio / sourceResolution,
+                      pixelRatio / sourceResolution);
+  stitchContext.translate(-srcDataExtent[0], srcDataExtent[3]);
+
+  goog.array.forEach(sources, function(src, i, arr) {
+    var xPos = src.extent[0];
+    var yPos = -src.extent[3];
+    var srcWidth = ol.extent.getWidth(src.extent);
+    var srcHeight = ol.extent.getHeight(src.extent);
+
+    if (wrapXType == ol.reproj.WrapXRendering_.STITCH_SHIFT) {
+      xPos = goog.math.modulo(xPos, wrapXShiftDistance);
+    }
+    stitchContext.drawImage(src.image, xPos, yPos, srcWidth, srcHeight);
+
+    if (wrapXType == ol.reproj.WrapXRendering_.STITCH_EXTENDED) {
+      stitchContext.drawImage(src.image, wrapXShiftDistance + xPos, yPos,
+                              srcWidth, srcHeight);
+    }
+  });
+
+  var targetTL = ol.extent.getTopLeft(targetExtent);
+
+  goog.array.forEach(triangulation.getTriangles(), function(tri, i, arr) {
```

Comment:
It would be possible, but the function heavily relies on the precalculated values, so a lot of variables would have to be passed.